File: java/oop/InterfaceApp/src/com/techlabs/icrudable/test/TestICRUDable.java
package com.techlabs.icrudable.test;

import com.techlabs.icrudable.CustomerDB;
import com.techlabs.icrudable.ICRUDable;
import com.techlabs.icrudable.InvoiceDB;
import com.techlabs.icrudable.VendorDB;

public class TestICRUDable {
	public static void main(String[] args) {
		ICRUDable icrudAble;
		
		icrudAble=new CustomerDB();
		toDoCreate(icrudAble);
		toDoUpdate(icrudAble);
		toDoRead(icrudAble);
		toDoDelete(icrudAble);
		
		icrudAble=new VendorDB();
		toDoCreate(icrudAble);
		toDoUpdate(icrudAble);
		toDoRead(icrudAble);
		toDoDelete(icrudAble);
		
		icrudAble=new InvoiceDB();
		toDoCreate(icrudAble);
		toDoUpdate(icrudAble);
		toDoRead(icrudAble);
		toDoDelete(icrudAble);
		
	}
	
	public static void toDoCreate(ICRUDable icrudAble){
		icrudAble.create();
	}
	
	public static void toDoUpdate(ICRUDable icrudAble){
		icrudAble.update();
	}
	
	public static void toDoRead(ICRUDable icrudAble){
		icrudAble.read();
	}
	
	public static void toDoDelete(ICRUDable icrudAble){
		icrudAble.delete();
	}
}

File: dotNet/CSharp/InterFaceCRUDApp/InterFaceCRUDApp/Test/TestICRUDAble.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace InterFaceCRUDApp.Test
{
    public class TestICRUDAble
    {
        static void Main(string[] args)
        {
            ICRUDAble iCRUDAble;

            iCRUDAble = new CustomerDB();
            iCRUDAble.Create();
            iCRUDAble.Update();
            iCRUDAble.Read();
            iCRUDAble.Delete();

            iCRUDAble = new VendorDB();
            iCRUDAble.Create();
            iCRUDAble.Update();
            iCRUDAble.Read();
            iCRUDAble.Delete();

            iCRUDAble = new InvoiceDB();
            iCRUDAble.Create();
            iCRUDAble.Update();
            iCRUDAble.Read();
            iCRUDAble.Delete();
        }

        public static void toDoCreate(ICRUDAble iCRUDAble)
        {
            iCRUDAble.Create();
        }

        public static void toDoUpdate(ICRUDAble iCRUDAble)
        {
            iCRUDAble.Update();
        }

        public static void toDoRead(ICRUDAble iCRUDAble)
        {
            iCRUDAble.Read();
        }

        public static void toDoDelete(ICRUDAble iCRUDAble)
        {
            iCRUDAble.Delete();
        }
    }
}
